framework/src/auxkernels/VariableTimeIntegrationAux.C

```diff
@@ -0,0 +1,94 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "VariableTimeIntegrationAux.h"
+
+template<>
+InputParameters validParams<VariableTimeIntegrationAux>()
+{
+  InputParameters params = validParams<AuxKernel>();
+  params.addRequiredCoupledVar("variable_to_integrate", "The variable to be integrated");
+  params.addParam<Real>("coefficient", 1.0, "A simple coefficient");
+  params.addParam<unsigned int>("order",2," the order of global truncation error, midpoint(1), trapazoidal(2),  simpson 1/3 rule (3)");
+  return params;
+}
+
+VariableTimeIntegrationAux::VariableTimeIntegrationAux(const InputParameters & parameters) :
+    AuxKernel(parameters),
+    _coef(getParam<Real>("coefficient")),
+    _order(getParam<unsigned int>("order"))
+{
+  Real constant = 1.0/3.0;
+
+  switch (_order)
+  {
+
+    case 1:
+      _integration_coef.push_back(1.0);
+      _coupled_vars.push_back(&coupledValue("variable_to_integrate"));
+      break;
+    case 2:
+      _integration_coef.push_back(0.5);
+      _integration_coef.push_back(0.5);
+      _coupled_vars.push_back(&coupledValue("variable_to_integrate"));
+      _coupled_vars.push_back(&coupledValueOld("variable_to_integrate"));
+      break;
+    case 3:
+      _integration_coef.push_back(constant);
+      _integration_coef.push_back(4.0*constant);
+      _integration_coef.push_back(constant);
+      _coupled_vars.push_back(&coupledValue("variable_to_integrate"));
+      _coupled_vars.push_back(&coupledValueOld("variable_to_integrate"));
+      _coupled_vars.push_back(&coupledValueOlder("variable_to_integrate"));
+      break;
+    default:
+      _integration_coef.push_back(0.5);
+      _integration_coef.push_back(0.5);
+      _coupled_vars.push_back(&coupledValue("variable_to_integrate"));
+      _coupled_vars.push_back(&coupledValueOld("variable_to_integrate"));
```

Comment:
I'd make the `default` case a mooseError("Must specify order"); or something similar.